### PR TITLE
Replace data_type_utils.cc throw calls

### PR DESCRIPTION
**Pull request by vinitra**
Follow on to #2888. Removing exception handling in ONNX Runtime is an important step for reducing the size for an ORT minimal build for mobile. This is a key deliverable of the ORT 1.5 release.

Short term fix approach: return an empty TypeProto and empty string short term instead of throwing errors.

For the TypeProto the caller can check:
`returned_type_proto.value_case() == TypeProto::VALUE_NOT_SET`

### CODE REVIEW

File: onnx/defs/data_type_utils.cc [hunk 1]
```diff
@@ -172,7 +173,7 @@ std::string DataTypeUtils::ToDataTypeString(int32_t tensor_data_type) {
   TypesWrapper& t = TypesWrapper::GetTypesWrapper();
   auto iter = t.TensorDataTypeToTypeStr().find(tensor_data_type);
   if (t.TensorDataTypeToTypeStr().end() == iter) {
-    throw std::invalid_argument("Invalid tensor data type.");
+    return "Invalid tensor data type.";
```
Comment (gwang-msft) [inline onnx/defs/data_type_utils.cc:176]: The 1st iter of the previous PR #2888 was trying to return an empty string, does the "Invalid tensor data type." actually mean anything to the caller?

File: onnx/defs/data_type_utils.cc [hunk 2]
```diff
@@ -112,7 +112,8 @@ const TypeProto& DataTypeUtils::ToTypeProto(const DataType& data_type) {
   std::lock_guard<std::mutex> lock(GetTypeStrLock());
   auto it = GetTypeStrToProtoMap().find(*data_type);
   if (GetTypeStrToProtoMap().end() == it) {
-    throw std::invalid_argument("Invalid data type.");
+    static const TypeProto empty_type_proto;
```
Comment (skottmckay) [inline onnx/defs/data_type_utils.cc:115]: Maybe add a comment to the header saying that if the input was invalid they will get an empty TypeProto back, and return_value.value_case() will return TypeProto::VALUE_NOT_SET.

File: onnx/defs/data_type_utils.cc (hunk 1 above)
Comment (gramalingam) [reply onnx/defs/data_type_utils.cc:176]: Agree. I see pros/cons, but for code that handles this, comparing against an empty string is simpler and easier.

Comment (gramalingam) [reply onnx/defs/data_type_utils.cc:176]: Or, a named string constant.

File: onnx/defs/data_type_utils.cc (hunk 2 above)
Comment (vinitra) [reply onnx/defs/data_type_utils.cc:115]: Thanks for the good suggestion! Addressed.

File: onnx/defs/data_type_utils.cc (hunk 1 above)
Comment (vinitra) [reply onnx/defs/data_type_utils.cc]: Resolved with an empty string.

File: onnx/defs/data_type_utils.cc [hunk 3]
```diff
@@ -109,10 +109,13 @@ DataType DataTypeUtils::ToType(const std::string& type_str) {
 }
 
 const TypeProto& DataTypeUtils::ToTypeProto(const DataType& data_type) {
+  // If the DataType input is invalid, this function will return an empty TypeProto
```
Comment (askhade) [inline onnx/defs/data_type_utils.cc:112]: Could you add this in the class header data_type_utils.h where this function is declared.

Comment (vinitra) [reply onnx/defs/data_type_utils.cc:112]: Updated!

**Review by askhade — APPROVED** [review]

**Review by gramalingam — APPROVED** [review]

**Pull request merged**